lib/longrunning.js

```diff
@@ -136,8 +136,18 @@ function Operation(grpcOp, longrunningDescriptor, backoffSettings) {
   this.completeListeners = 0;
   this.hasActiveListeners = false;
 
-  this.currentOperation = grpcOp;
+  this.latestResponse = grpcOp;
   this.longrunningDescriptor = longrunningDescriptor;
+  this.result = null;
+  if (grpcOp.response && grpcOp.response.value) {
+    this.result =
+      this.longrunningDescriptor.responseDecoder(grpcOp.response.value);
+  }
+  this.metadata = null;
+  if (grpcOp.metadata && grpcOp.metadata.value) {
+    this.metadata =
+      this.longrunningDescriptor.metadataDecoder(grpcOp.metadata.value);
+  }
```

Comment:
This code block is similar to the one in `unpackResponse` in `getOperation`. Can we consolidate the implementation by introducing a new method `Operation.prototype._unpackResponse`?